src/OrbitGl/TimerTrack.cpp

```diff
@@ -251,6 +251,13 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
 
   for (auto& chain : chains_by_depth) {
     if (!chain) continue;
+    // In order to draw overlaps correctly, we need for every text box to be drawn (current),
+    // its previous and next text box. In order to avoid looking ahead for the next text (which is
+    // error-prone), we are doing just one traversal of the text boxes, while keeping track of the
+    // previous two timers and will draw the previous one, thus the currents iteration value being
```

Comment:
// error-prone), we are doing just one traversal of the text boxes, while keeping track of the
    // previous two timers, thus the currents iteration value being the "next" textbox.

Reply:
Done.